Makefile.rules

```diff
@@ -0,0 +1,162 @@
+install-shared-%: BASENAME=$(patsubst install-shared-%,%,$@)
+install-shared-%:
+	mkdir -p $(DESTDIR)$(libdir)/
+	$(INSTALL) -m 755 -v $(BASENAME).so.$($(BASENAME)_VERSION) $(DESTDIR)$(libdir)/
+	$(RM) -rf $(DESTDIR)$(libdir)/$(BASENAME).so $(DESTDIR)$(libdir)/$(BASENAME).so.$($(BASENAME)_SOVERSION)
+	ln -sf $(BASENAME).so.$($(BASENAME)_VERSION) $(DESTDIR)$(libdir)/$(BASENAME).so
+	ln -sf $(BASENAME).so.$($(BASENAME)_VERSION) $(DESTDIR)$(libdir)/$(BASENAME).so.$($(BASENAME)_SOVERSION)
+
+################################################################################
+# Pseudo-rules for maintaining development build tree
+
+clean:
+	find $(BUILD_OUT_PREFIX). -name "*.[ao]" -o -name "*.l[ao]" -o -name "*.so" -o -name "*.so.*" -o -name "*.gcda" -o -name "*.gcno" | xargs rm -f
+	rm -f $(GENERATED_FILES)
+
+distclean: clean depclean
+	rm -rf config.log config.status include/tscb/config Makefile .depend autom4te.cache $(PKGCONFIG_FILES)
+
+maintainerclean: distclean
+	rm -f configure
+
+# dependency rules
+
+$(BUILD_OUT_PREFIX).dep/%.la.d: %.cpp
+	@mkdir -p $(dir $@)
+	@$(CXX) -MM $(CXXFLAGS) $(CPPFLAGS) -MT $(patsubst %.cpp, $(BUILD_OUT_PREFIX)%.la, $<) -MP -MF $@ $<
+	@echo MAKEDEP $<
+
+$(BUILD_OUT_PREFIX).dep/%.lo.d: %.cpp
+	@mkdir -p $(dir $@)
+	@$(CXX) -MM -DPIC -fPIC $(CXXFLAGS) $(CPPFLAGS) -MT $(patsubst %.cpp, $(BUILD_OUT_PREFIX)%.lo, $<) -MP -MF $@ $<
+
+depclean:
+	rm -rf $(BUILD_OUT_PREFIX).dep
+
+DEPEND = $(patsubst %.cpp, $(BUILD_OUT_PREFIX).dep/%.la.d, $(SOURCES)) $(patsubst %.cpp, $(BUILD_OUT_PREFIX).dep/%.lo.d, $(SOURCES))
+depend: $(DEPEND)
+ifeq ($(shell if [ -e $(BUILD_OUT_PREFIX).dep ] ; then echo yes ; fi),yes)
+-include $(DEPEND)
+endif
+
+echo-depends:
+	echo $(DEPEND)
+
+################################################################################
+# Unit testing rules
+
+SOURCES+=$(patsubst %, %.cpp, $(TESTS))
+RUNTESTS=$(patsubst %, run-%, $(TESTS))
+VALGRINDTESTS=$(patsubst %, valgrind-%, $(TESTS))
+
+$(TESTS): % : %.la
+$(TESTS): LDFLAGS += -pthread
+
+$(RUNTESTS): run-% : %
+	$^
+
+$(VALGRINDTESTS): valgrind-% : %
+	valgrind --leak-check=full --error-exitcode=1 $^
+
+check: $(RUNTESTS)
+
+valgrind-check: $(VALGRINDTESTS)
+
+################################################################################
+# Unit test coverage rules
+
+$(BUILD_OUT_PREFIX)%.coverage.la: %.cpp
+	$(CXX) -c $(CXXFLAGS) $(CPPFLAGS) $(COVERAGEFLAGS) -o $@ $<
+
+$(COVERAGE_TESTS): $(BUILD_OUT_PREFIX)%.coverage : $(BUILD_OUT_PREFIX)%.coverage.la
+$(COVERAGE_TESTS): LDFLAGS += -pthread $(COVERAGEFLAGS)
+
+coverage: $(COVERAGE_TESTS) .PHONY
+	for CTEST in $(COVERAGE_TESTS) ; do $$CTEST ; done
+	mkdir -p $(BUILD_OUT_PREFIX)coverage
+	gcovr . --html-details -o $(BUILD_OUT_PREFIX)coverage/coverage.html
+
+################################################################################
+# Documentation rules
+
+docclean:
+	rm -rf doc
+
+doc: .PHONY
+	mkdir -p doc
+	doxygen doxygen.conf
```

Comment:
This command creates the doc folder, but then writes the documentation in the docs folder. Either the rule name, or the `OUTPUT_DIRECTORY` variable in the doxygen.conf needs to be adjusted.

Reply:
changed target name